### PR TITLE
imlementing pdp

### PR DESCRIPTION
**Pull request by batchatT**

### CODE REVIEW

File: app/src/main/res/layout/fragment_product_details_page.xml [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+            app:layout_constraintTop_toTopOf="@id/product_view" />
+
+        <androidx.appcompat.widget.AppCompatRatingBar
+            android:id="@+id/rating_bar"
+            style="?android:attr/ratingBarStyleSmall"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="@dimen/margin32"
+            android:layout_marginTop="@dimen/margin12"
+            android:numStars="5"
```
Comment (denisque75) [inline app/src/main/res/layout/fragment_product_details_page.xml:77]: may be is it better to set it in your style.xml?

File: app/src/main/res/layout/fragment_product_details_page.xml [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+            android:rating="@{viewModel.product.rating.rate}"
+            android:stepSize="0.1"
```
Comment (denisque75) [inline app/src/main/res/layout/fragment_product_details_page.xml:79]: may be is it better to set it in your style.xml?

File: app/src/main/res/layout/fragment_profile_page.xml [hunk 3]
```diff
@@ -1,21 +1,21 @@
 <?xml version="1.0" encoding="utf-8"?>
 <layout xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:tools="http://schemas.android.com/tools"
-    xmlns:app="http://schemas.android.com/apk/res-auto">
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools">
 
     <androidx.constraintlayout.widget.ConstraintLayout
```
Comment (denisque75) [inline app/src/main/res/layout/fragment_profile_page.xml:6]: Use linearLayout

File: app/src/main/java/com/example/trempelapp/presentation_layer/mainScreen/main_fragments/product/ProductListViewModel.kt [hunk 4]
```diff
@@ -17,6 +16,10 @@ class ProductListViewModel @Inject constructor(
         get() = _productListLiveData
     private val _productListLiveData = SingleLiveEvent<List<Product>>()
 
+    val onProductClickedLiveData: SingleLiveEvent<Boolean>
```
[the remark this answers is not in the document]
Comment (batchatT) [reply app/src/main/java/com/example/trempelapp/presentation_layer/mainScreen/main_fragments/product/ProductListViewModel.kt:19]: done

File: app/src/main/java/com/example/trempelapp/presentation_layer/mainScreen/main_fragments/product/ProductListViewModel.kt [hunk 5]
```diff
@@ -40,11 +43,14 @@ class ProductListViewModel @Inject constructor(
...
 
+    lateinit var productItem: Product
+
     val adapter by lazy {
         ProductRecyclerAdapter().apply {
             setOnProductListener(object : ProductRecyclerAdapter.OnProductListener {
                 override fun onCategoryListener(product: Product) {
-                    Log.d("recycler", "item clicked")
+                    productItem = product
+                    _onProductClickedLiveData.value = true
```
[the remark this answers is not in the document]
Comment (batchatT) [reply app/src/main/java/com/example/trempelapp/presentation_layer/mainScreen/main_fragments/product/ProductListViewModel.kt:53]: done

File: app/src/main/java/com/example/trempelapp/utils/BindingUtils.kt [hunk 6]
```diff
@@ -23,20 +25,34 @@ fun setVisibility(progressBar: ProgressBar, isVisible: Boolean) {
     }
 }
 
+private const val TAG = "IMAGE"
 @BindingAdapter("imageUrl")
 fun loadImage(view: ImageView, url: String?) {
-    Glide
+//    Glide
```
[the remark this answers is not in the document]
Comment (batchatT) [reply app/src/main/java/com/example/trempelapp/utils/BindingUtils.kt:31]: done

File: app/src/main/res/layout/fragment_cart_page.xml [hunk 7]
```diff
@@ -1,21 +1,21 @@
 <?xml version="1.0" encoding="utf-8"?>
 <layout xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:tools="http://schemas.android.com/tools"
-    xmlns:app="http://schemas.android.com/apk/res-auto">
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools">
 
     <androidx.constraintlayout.widget.ConstraintLayout
```
[the remark this answers is not in the document]
Comment (batchatT) [reply app/src/main/res/layout/fragment_cart_page.xml:6]: done

File: app/src/main/res/layout/fragment_product_details_page.xml (hunk 1 above)
Comment (batchatT) [reply app/src/main/res/layout/fragment_product_details_page.xml:77]: done

File: app/src/main/res/layout/fragment_product_details_page.xml (hunk 2 above)
Comment (batchatT) [reply app/src/main/res/layout/fragment_product_details_page.xml:79]: done

File: app/src/main/res/layout/fragment_profile_page.xml (hunk 3 above)
Comment (batchatT) [reply app/src/main/res/layout/fragment_profile_page.xml:6]: done

File: app/src/main/java/com/example/trempelapp/presentation_layer/mainScreen/main_fragments/pdp/ProductDetailsPageFragment.kt [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    private val binding by lazy {
+        FragmentProductDetailsPageBinding.inflate(layoutInflater)
+    }
+
+    override fun handleArguments() {
+        super.handleArguments()
+        arguments?.let {
+            viewModel.product = it.getParcelable(PRODUCT_TO_DETAILS_KEY)!!
```
[the remark this answers is not in the document]
Comment (batchatT) [reply app/src/main/java/com/example/trempelapp/presentation_layer/mainScreen/main_fragments/pdp/ProductDetailsPageFragment.kt]: done

**Pull request merged**